upgrade-1.x-to-2.x.sh

```diff
@@ -0,0 +1,410 @@
+    log "Host OS version $VERSION is less than $PREFERRED_HOSTOS_VERSION, installing tools..."
+    tools_path=/home/root/upgrade_tools
+    mkdir -p $tools_path
+    export PATH=$tools_path:$PATH
+    case $BINARY_TYPE in
+	arm)
+	    download_uri=https://github.com/resin-os/resinhup/raw/onetotwo/upgrade-binaries/$BINARY_TYPE
+	    curl -s -L -o $tools_path/tar $download_uri/tar
+	    chmod 755 $tools_path/tar
+	    curl -s -L -o $tools_path/e2label $download_uri/e2label
+	    chmod 755 $tools_path/e2label
+	    curl -s -L -o $tools_path/mkfs.ext4 $download_uri/mkfs.ext4
+	    chmod 755 $tools_path/mkfs.ext4
+	    curl -s -L -o $tools_path/resize2fs $download_uri/resize2fs
+	    chmod 755 $tools_path/resize2fs
+	    ;;
+	*)
+	    log ERROR "Binary type $BINARY_TYPE not supported."
+	    ;;
+    esac
+fi
+
+# Stop docker containers, /var/lib/docker and /resin-data on p6
+log "Stopping all containers..."
+systemctl stop update-resin-supervisor.timer > /dev/null 2>&1
+systemctl stop resin-supervisor > /dev/null 2>&1
+$DOCKER stop $($DOCKER ps -a -q) > /dev/null 2>&1 || true
+log "Removing all containers..."
+$DOCKER rm $($DOCKER ps -a -q) > /dev/null 2>&1 || true
+systemctl stop docker
+
+# Switch connman to use bind mounted state dir
+log "Switching connman to bind mounted state dir..."
+mkdir -p /tmp/connman
+cp -a /var/lib/connman/* /tmp/connman/
+# Versions before 1.20 need this to prevent dropping VPN
+sed -i -e 's/NetworkInterfaceBlacklist=docker,veth,tun,p2p/NetworkInterfaceBlacklist=docker,veth,tun,p2p,resin-vpn/' /etc/connman/main.conf
+mount -o bind /tmp/connman /var/lib/connman
+systemctl restart connman
+
+# Save /resin-data to rootB
+mkfs.ext4 -F ${root_dev}p3
+mkdir -p /tmp/backup
+mount ${root_dev}p3 /tmp/backup
+(cd /mnt/data; tar -zcf /tmp/backup/resin-data.tar.gz resin-data)
+
+# Unmount p6
+log "Unmounting filesystems..."
+umount /mnt/data
+umount /var/lib/docker
+umount /resin-data
+
+# Save conf contents to /boot
+# XXX
+
+# Unmount p5 if mounted
+if mount | grep /mnt/conf; then
+    umount /mnt/conf
+fi
+
+# Unmount p1
+umount /mnt/boot
+
+# Delete partitions 4-6
+parted -s $root_dev rm 6
+parted -s $root_dev rm 5
+parted -s $root_dev rm 4
+
+# Create extended partition
+parted -s $root_dev mkpart extended 700 100%
+
+# Create state partition
+parted -s $root_dev mkpart logical ext4 705 726
+
+# Create data partition (using type btrfs corrupts partition table)
+parted -s $root_dev mkpart logical ext4 730 100%
+
+# Create resin-state filesystem
+mkfs.ext4 -F -L resin-state ${root_dev}p5
+
+# Create resin-data filesystem
+mkfs.btrfs -f -L resin-data ${root_dev}p6
+
+# Remount data partition
+mount ${root_dev}p6 /mnt/data
+
+# Copy backup of resin-data
+cp /tmp/backup/resin-data.tar.gz /mnt/data
+
+# Unmount backup
+umount /tmp/backup
+
+parted -s $root_dev rm 3
+
+# Resize partition 2 to 327MB
+parted -s $root_dev resizepart 2 373.1
```

Comment:
Can have some comment/calculation here, how the value 373.1 correspond to that resize that the comment above mentions?